dist/init/linux-upstart/caddy.conf

```diff
@@ -1,18 +1,25 @@
-description "Caddy startup script"
+description "Caddy HTTP/2 web server"
 author "Mathias Beke"
 
 start on runlevel [2345]
 stop on runlevel [016]
 
+console log
 
 setuid www-data
 setgid www-data
 
 respawn
 respawn limit 10 5
 
-limit nofile 4096 4096
+# Letsencrypt-issued certificates will be written to this directory.
+env HOME=/etc/ssl/caddy
+
+limit nofile 1048576 1048576
 
 script
-	exec /usr/bin/caddy -agree=true -conf=/etc/caddy/Caddyfile
-end script
\ No newline at end of file
+        mkdir -p /etc/caddy
+        rootdir=`mktemp -d -t "caddy-run.XXXXXX"`
```

Comment:
If it's a *bash* script (I don't know that) then use `$()`. The value should be quoted.

```bash
rootdir="$(mktemp -d -t "…")"
```

… Is there a way to run something when the process exits? If so, remove that directory there.

One more reason to like *systemd's* `PrivateTmp`. :-)